### PR TITLE
Can if containers

### PR DESCRIPTION
**Pull request by moh1139**
Made Modifications to follow our last meeting notes.
Meeting Notes:
1- Add word to the suffix of each struct name, to follow the AUTOSAR naming convention

2- All #defines are in UpperCase, whether it's a pre-compile parameter or not, and with underscores between each word (after the module name)
Example: _DISPATCH_USER_CHECK_TRCV_WAKE_FLAG_INDICATION_UL

3- Variables are in LowerCase with it's first letter in UpperCase

4- Data type of enum paramters in the structure has to be the same as the defined in the Cfg file
Example:
typedef uint8 CANIF_DISPATCH_USER_CTRL_BUS_OFF_UL;
#define CAN_SM ((CANIF_DISPATCH_USER_CTRL_BUS_OFF_UL)0x00)
#define CDD ((CANIF_DISPATCH_USER_CTRL_BUS_OFF_UL)0x01)
typedef struct{
CANIF_DISPATCH_USER_CTRL_BUS_OFF_UL CanIfDispatchUserCtrlBusOffUL;
}CanIfDispatchCfgType;

5- Function names that will be passed, should be without ()
Example:
#if(CANIF_DISPATCH_USER_CTRL_BUS_OFF_UL==CAN_SM)
#define CANIF_DISPATCH__USER_CTRL_BUS_OFF_NAME CanSM_ControllerBusOff /* no ( ) at the end of the function name /
#else if(CANIF_DISPATCH_USER_CTRL_BUS_OFF_UL==CDD)
#define CANIF_DISPATCH__USER_CTRL_BUS_OFF_NAME Cdd_ControllerBusOff / no ( ) at the end of the function name */
#endif

6- PostBuild paramaters that has variables in the struct don't need to make a #define for its init value, it's value will be assigned from the PostBuild file (in the struct object)

Side Note:

PostBuild paramaters can be #defines only without structs.....depends on what we agreed for implementation

### CODE REVIEW

File: Software/bsw/gen/CanIf_Cfg.h [hunk 1]
```diff
@@ -245,86 +245,86 @@ typedef uint16 CanIfPublicHandleTypeEnum;
...
 
 /* Identifier mask which denotes relevant bits in the CAN Identifier. This
 parameter may be used to keep parts of the CAN Identifier of dynamic
 transmit L-PDUs static. Range: 11 bits for Standard CAN Identifier, 29
 bits for Extended CAN Identifier.
 Range: 0 .. 536870911 */
-#define CanIfTxPduCanIdMask0					(1<<(uint32)0)
-#define CanIfTxPduCanIdMask1					(1<<(uint32)1)
+#define CANIF_TX_PDU_CAN_ID_MASK0				(1<<(uint32)0)
+#define CANIF_TX_PDU_CAN_ID_MASK1				(1<<(uint32)1)
```
Comment (autosarzs) [inline Software/bsw/gen/CanIf_Cfg.h:279]: What the using of this mask, if to check is this id extended or normal I believe should be not 1

File: Software/bsw/gen/CanIf_Cfg.h [hunk 2]
```diff
@@ -245,86 +245,86 @@ typedef uint16 CanIfPublicHandleTypeEnum;
...
-#define CanIfMaxTxPduCfgValue					(18446744073709551615)
+#define CANIF_MAX_TX_PDU_CFG					(18446744073709551615)
 
 /* CAN Identifier of transmit CAN L-PDUs used by the CAN Driver for
 CAN L-PDU transmission. Range: 11 Bit For Standard CAN Identifier
 ... 29 Bit For Extended CAN identifier
 The CAN Identifier may be omitted for dynamic transmit L-PDUs. */
-#define CanIfTxPduCanId0						(0U)
-#define CanIfTxPduCanId1						(1U)
+#define CANIF_TX_PDU_CAN_ID0					(0U)
```
Comment (autosarzs) [inline Software/bsw/gen/CanIf_Cfg.h:270]: The same here, I believe it’s not 0 and 1

**Review by autosarzs — CHANGES_REQUESTED** [review]

**Comment by ahmedwahdan** [comment]
Pull request changes kept for long time will be hard to follow and review, please push for review and merge.

**Comment by moh1139** [comment]
> Pull request changes kept for long time will be hard to follow and review, please push for review and merge.

I was waiting for the review results, in order to merge or to make modifications if any, and review results was just made 2 hours ago

File: Software/bsw/gen/CanIf_Cfg.h (hunk 1 above)
Comment (moh1139) [reply Software/bsw/gen/CanIf_Cfg.h:279]: CanIfTxPduCanIdType parameter is the one that checks if its EXTENDED_CAN, EXTENDED_FD_CAN, STANDARD_CAN, STANDARD_FD_CAN.
not CanIfTxPduCanIdMask nor CanIfTxPduCanId.
what i understood from SWS is that they are used for MetaData. in SWS Page 35 (7.4 Dynamic L-PDUs).

and the below text is what in SWS in for the Dynamic L-PDUs:

CanIf shall support the ability to filter incoming messages using the CanIfRxPdu-
CanIdMask. The filtering shall be done by comparing the incoming CanId with the
stored CanIfRxPduCanId after applying the CanIfRxPduCanIdMask to both IDs.
This should be done after the filtering of regular CanIds without mask, to allow for
separate handling of some of the CanIds that fall into the range defined by the mask
or a CanId based range.
During transmission of dynamic L-SDUs, when a CanIfTxPduCanIdMask is defined,
the variable parts of the CanId provided via the MetaData must be merged with the
CanId by using this mask. When no CanIfTxPduCanIdMask and no CanIfTxPdu-
CanId are configured, the MetaData shall be used directly as CanId.

File: Software/bsw/gen/CanIf_Cfg.h (hunk 2 above)
Comment (moh1139) [reply Software/bsw/gen/CanIf_Cfg.h:270]: i made 0 and 1 for multiplicity only and for the parameter usage is the same as i mentioned above in the Mask:
CanIfTxPduCanIdType parameter is the one that checks if its EXTENDED_CAN, EXTENDED_FD_CAN, STANDARD_CAN, STANDARD_FD_CAN.
not CanIfTxPduCanIdMask nor CanIfTxPduCanId.
what i understood from SWS is that they are used for MetaData. in SWS Page 35 (7.4 Dynamic L-PDUs).

and the below text is what in SWS in for the Dynamic L-PDUs:

CanIf shall support the ability to filter incoming messages using the CanIfRxPdu-
CanIdMask. The filtering shall be done by comparing the incoming CanId with the
stored CanIfRxPduCanId after applying the CanIfRxPduCanIdMask to both IDs.
This should be done after the filtering of regular CanIds without mask, to allow for
separate handling of some of the CanIds that fall into the range defined by the mask
or a CanId based range.
During transmission of dynamic L-SDUs, when a CanIfTxPduCanIdMask is defined,
the variable parts of the CanId provided via the MetaData must be merged with the
CanId by using this mask. When no CanIfTxPduCanIdMask and no CanIfTxPdu-
CanId are configured, the MetaData shall be used directly as CanId.

**Comment by moh1139** [comment]
New Modification:
Modified the TypeDef definition to follow AUTOSAR naming Convention i.e: Add word "Type" in the suffix of the enum parameter name

File: Software/bsw/static/COM/CanIf/inc/CanIf.h [hunk 3]
```diff
@@ -376,7 +382,7 @@ typedef struct {
 
 				/*This container contains the configuration (parameters) of an adressed CAN controller by
 				an underlying CAN Driver module. This container is configurable per CAN controller.*/	
-				CanIfCtrlCfgType CanIfCtrlCfgObj;
+				CanIfCtrlCfgType CanIfCtrlCfgObj;//It should be CanIfCtrlCfgType CanIfCtrlCfgObj[MAX_CONFIGURABLE_CAN_CONTROLLERS];
```
Comment (Abdelrahman1412) [inline Software/bsw/static/COM/CanIf/inc/CanIf.h]: Yes and the naming of MAX_CONFIGURABLE_CAN_CONTROLLERS must match the one in CAN DRIVER

**Review by Abdelrahman1412 — APPROVED** [review]

File: Software/bsw/static/COM/CanIf/inc/CanIf.h (hunk 3 above)
Comment (moh1139) [reply Software/bsw/static/COM/CanIf/inc/CanIf.h]: Sure InshaAllah

File: Software/bsw/gen/CanIf_Cfg.h (hunk 1 above)
Comment (autosarzs) [reply Software/bsw/gen/CanIf_Cfg.h:279]: You’re correct but how the CAN driver know about this ID is FD or standards or other ? We are pul a mask at the end of the id and let the CAN check this using that mask but anyway let us move forward without FD and Extended support

**Review by autosarzs — APPROVED** [review]
